src/tools/tidy/src/deps.rs

```diff
@@ -486,73 +468,55 @@ fn check_permitted_dependencies(
     restricted_dependency_crates: &[&'static str],
     bad: &mut bool,
 ) {
+    let mut deps = HashSet::new();
+    for to_check in restricted_dependency_crates {
+        let to_check = pkg_from_name(metadata, to_check);
+        use cargo_platform::Cfg;
+        use std::str::FromStr;
+        // We don't expect the compiler to ever run on wasm32, so strip
+        // out those dependencies to avoid polluting the permitted list.
+        deps_of_filtered(metadata, &to_check.id, &mut deps, &|dep_kinds| {
+            dep_kinds.iter().any(|dep_kind| {
+                dep_kind
+                    .target
+                    .as_ref()
+                    .map(|target| {
+                        !target.matches(
+                            "wasm32-unknown-unknown",
+                            &[
+                                Cfg::from_str("target_arch=\"wasm32\"").unwrap(),
+                                Cfg::from_str("target_os=\"unknown\"").unwrap(),
+                            ],
+                        )
```

Comment:
One slight concern here: I think cargo-vendor will still pick these dependencies up (and that's necessary for building std on wasm32), but if there's e.g. a GPL dependency on wasm that's probably still something we want to be careful about distributing/including into our tarballs, even if it's technically not used at all, right?

Is there a way to filter targets for cargo vendor? I assume "no"...

Reply:
Oh I see this was already asked and answered... never mind.